fog/view/server/src/block_tracker.rs

```diff
@@ -90,6 +91,38 @@ where
         self.sharding_strategy.should_process_block(block_index)
     }
 
+    /// Given a requested `block_index` to start from, and a number of consecutive blocks
+    /// (`block_count`), return a BlockRange that is contained with this shard's `ShardingStrategy`.
+    pub fn get_in_bounds_block_range(
+        &self,
+        block_index: u64,
+        block_count: usize,
+    ) -> Result<BlockRange, String> {
+        if !self
+            .sharding_strategy
+            .get_block_range()
+            .contains(block_index)
+        {
+            return Err(format!(
```

Comment:
Yeah, I thought about this when designing it. The thing to keep in mind is that the block index of 200 will eventually be supplied to this function because the caller is going sequentially through each block. I figured it would be cleaner to only process blocks if the provided index is in range. I think this because the query batch size is really a greedy attempt to load more blocks, and I think it would be confusing if we load more blocks for a block index that isn't in range.